s30900-Pawel_Kozlowski/Zadania/18.11.2023 Kolokwium powtórzenie Zadania.txt

```diff
@@ -0,0 +1,115 @@
+Zadania do wykonania przed kolokwium wykonane ze zrozumieniem
+
+1. Jest dane takie drzewo katalogów:
+
+   .
+
+   ├── ala
+
+   │   ├── i
+
+   │   │   └── as
+
+   │   └── ma
+
+   │       └── kota
+
+   └── kot
+   
+       └── ma
+
+           └── ale
+
+   Znajdujesz się w katalogu kota. Katalog ala, jest w katalogu /tmp/ukos.
+   Jak przejść do katalogu ala za pomocą:
+
+   ścieżki względnej (relatywnej) -> cd ../.. (umiem)
+
+   ścieżki bezwzględnej (absolutnej)} -> cd /tmp/ukos/ala (umiem)
+
+2. Masz taki układ katalogów, jak wyżej i ciągle jesteś w katalogu kota. 
+   Jak utworzyć poddrzewo katalogów jan/kowalski w katalogu ale za pomocą jednej komendy?
+
+   Podaj rozwiązania używając ścieżek absolutnych i relatywnych. (umiem proste)
+
+
+   relatywnych -> mkdir -p jan/kowalski ../../../kot/ma/ale
+
+   absolutnych -> mkdir -p /tmp/ukos/kot/ma/ale/jan/kowalski
+
+3. Masz taki układ katalogów, jak wyżej i ciągle jesteś w katalogu kota. Jak przenieść katalog ale do katalogu i używając:
+
+   źródło (ścieżka względna), miejsce docelowe (ścieżka absolutna)
+   -> mv ../../../kot/ma/ale /tmp/ukos/ala/i
+
+   źródło (ścieżka absolutna), miejsce docelowe (ścieżka względna)
+   -> mv /tmp/ukos/kot/ma/ale ../../i
+
+4. Jak zamknąć program, który nie reaguje na ctrl+c ?
+
+   Ctrl i + z lub kill %1
+
+5. Jak wypisać wszystkie pliki (i tylko pliki bez katalogów) w bieżącym katalogu, których nazwa zaczyna się od al?
+   Tu będzie trochę opisu ode mnie na temat wyrażeń regularnych.
+
+   -> ls al*
+
+6a. Nadaj uprawnienia do katalogu ala tak aby:
+   każdy mógł do niego wejść
+   tylko grupa mogła wyświetlić co w nim jest
+   właściciel ma pełne uprawnienia
+   HINT: Jakie są uprawnienia początkowe?
+   Podpowiedź: powyższe 3 punkty opisu to jedno zadanie. Nie 3 osobne.
+   Przygotuj 3 wariacje rozwiązania tego zadania używając różnych notacji: 
+   Wszystkie wersje powinny być zapisane w jednym poleceniu (bez używania && czy | )
+
+   oktalnej/ósemkowej/cyfrowej
+   -> chmod 751 /tmp/ukos/ala
+
+   literowej ze znakiem równości
+   -> chmod a=rwx,g=rx,o=x /tmp/ukos/ala
```

Comment:
`a=rwx` tu zadziała ale skoro osobno przypisuje uprawnienia grupy i innych to wtedy tu raczej powinna być użyta inna literka niż `a`